src/sas/sascalc/data_util/qsmearing.py

```diff
@@ -35,18 +39,42 @@ def smear_selection(data, model = None):
     """
     # Sanity check. If we are not dealing with a SAS Data1D
     # object, just return None
+    # This checks for 2D data (does not throw exception because fail is common)
     if  data.__class__.__name__ not in ['Data1D', 'Theory1D']:
         if data == None:
             return None
         elif data.dqx_data == None or data.dqy_data == None:
             return None
-        return PySmear2D(data, model)
-
+        return Pinhole2D(data)
+    # This checks for 1D data with smearing info in the data itself (again, fail is likely; no exceptions)
     if  not hasattr(data, "dx") and not hasattr(data, "dxl")\
          and not hasattr(data, "dxw"):
         return None
 
     # Look for resolution smearing data
+    # This is the code that checks for SESANS data; it looks for the file loader
+    # TODO: change other sanity checks to check for file loader instead of data structure?
+    _found_sesans = False
+    #if data.dx is not None and data.meta_data['loader']=='SESANS':
+    if data.dx is not None and data.isSesans:
+        #if data.dx[0] > 0.0:
+        if numpy.size(data.dx[data.dx <= 0]) == 0:
+            _found_sesans = True
+        # if data.dx[0] <= 0.0:
+        if numpy.size(data.dx[data.dx <= 0]) > 0:
+            raise ValueError('one or more of your dx values are negative, please check the data file!')
+
+    if _found_sesans == True:
+        #Pre-compute the Hankel matrix (H)
+        qmax, qunits = data.sample.zacceptance
+        hankel = sesans.SesansTransform()
+        sesans.SesansTransform.set_transform(hankel,
+        SE = Converter(data._xunit)(data.x, "A"),
+        zaccept = Converter(qunits)(qmax, "1/A"),
+        Rmax = 10000000)
+        # Then return the actual transform, as if it were a smearing function
+        return PySmear(SESANS1D(data, hankel._H0, hankel._H, hankel.q), model)
```

Comment:
I tried that before making these 2 classes.
SESANS1D inherit from Sesanstransform and adds some stuff to make it a Resolution object. 
This can then be used without decreasing the efficiency of the Sesanstransform calculation.
Merging the classes reduces efficiency (I'm sure there's a way to merge them without this loss, but I don't know how...)